src/__tests__/suggestions.js

```diff
@@ -0,0 +1,175 @@
+import {configure} from '../config'
+import {screen} from '..'
+import {renderIntoDocument} from './helpers/test-utils'
+
+beforeEach(() => {
+  configure({showSuggestions: true})
+})
+it('should not suggest when using getByRole', () => {
+  renderIntoDocument(`<button data-testid="foo">submit</button>`)
+
+  expect(() => screen.getByRole('button', {name: /submit/})).not.toThrowError()
+})
+
+test('should not suggest when nothing available', () => {
+  renderIntoDocument(`<span data-testid="foo" />`)
+
+  expect(() => screen.queryByTestId('foo')).not.toThrowError()
+})
+
+test(`should not suggest if the suggestion would give different results`, () => {
+  renderIntoDocument(`
+    <input type="text" data-testid="foo" /><span data-testid="foo" />
+  `)
+
+  expect(() => screen.getAllByTestId('foo')).not.toThrowError()
+})
+
+test('should suggest getByRole when used with getBy', () => {
+  renderIntoDocument(`<button data-testid="foo">submit</button>`)
+
+  expect(() => screen.getByTestId('foo')).toThrowErrorMatchingInlineSnapshot(`
+"A better query is available, try this:
+*ByRole("button", {name:/submit/})
```

Comment:
@kentcdodds I wanted to do that, but it would require having the make*Query functions pass down the first bit of the query.  if you want i can try that, but it was gonna touch a lot of files so I held off...

Reply:
The reason it would touch a lot of files is because they all end up going through the `queryAllBy` so afaict it would mean having to pass `get|find|query` w/ each make*Query call.

Reply:
@kentcdodds,  got this working last night.